app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantListViewModel.kt

```diff
@@ -35,26 +42,37 @@ class PlantListViewModel @ViewModelInject internal constructor(
     @Assisted private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
-    val plants: LiveData<List<Plant>> = getSavedGrowZoneNumber().switchMap {
-        if (it == NO_GROW_ZONE) {
+    private val growZone = MutableStateFlow(NO_GROW_ZONE)
+
+    val plants: LiveData<List<Plant>> = growZone.flatMapLatest { zone ->
+        if (zone == NO_GROW_ZONE) {
             plantRepository.getPlants()
         } else {
-            plantRepository.getPlantsWithGrowZoneNumber(it)
+            plantRepository.getPlantsWithGrowZoneNumber(zone)
         }
+    }.asLiveData()
+
+    init {
+        getSavedGrowZoneNumberOrDefault()
     }
 
     fun setGrowZoneNumber(num: Int) {
+        growZone.value = num
         savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, num)
     }
 
     fun clearGrowZoneNumber() {
+        growZone.value = NO_GROW_ZONE
         savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, NO_GROW_ZONE)
```

Comment:
These `savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, zone)` could be simplified if we use the power of flow operators. What about using `.onEach` for that?

```
val plants: LiveData<List<Plant>> = growZone.flatMapLatest { zone ->
        // rest of the code
    }.onEach {
        savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, it)
    }
     .asLiveData()
```

Reply:
Spoke offline - let's try moving the `.set` logic to the `init` block.